doc/userguide/rule-management/rule-reload.rst

```diff
@@ -1,32 +1,70 @@
 Rule Reloads
 ============
 
-Suricata can reload the rules without restarting. This way, there
-is minimal service disruption.
+Suricata was designed to reload rules while it is actively processing
+network traffic to minimize service disruption.
 
-This works by sending Suricata a signal or by using the unix socket. When Suricata is told to reload the rules these are the basic steps it takes:
+Suricata must be administratively directed to reload rules while it is running; there
+are two ways to cause this:
 
-* Load new config to update rule variables and values.
-* Load new rules
-* Construct new detection engine
-* Swap old and new detection engines
-* Make sure all threads are updated
-* Free old detection engine
+* Via a UNIX domain socket with the command ``reload-rules`` or ``ruleset-reload-nonblocking``
+* Upon receipt of the ``USR2`` signal
 
-Suricata will continue to process packets normally during this process. Keep in mind though, that the system should have enough memory for both detection engines.
+Suricata performs these steps while reloading rules:
 
-Signal::
+* The main Suricata configuration is reloaded to update rule variables and values; this
+  includes the rule related files ``classification.config``, ``reference.config`` and
+  ``theshold.config``
+* All rule files are reloaded with new rule variables applied
+* A new detection engine is created for the updated rules.
+* The previous and newly created detection engines are swapped.
+* Ensure all threads are updated
+* Free old detection engine and associated resources.
```

Comment:
This looks devguide worthy to me. Not so sure if this is of relevance to the user. wdyt?

Reply:
I think something is useful here for end-users, like how long and the is the old detection engine freed? At the end of the flows?  Immediately when the new one is ready?

Reply:
Maybe we could either add it to the Devguide (somewhere under `Suricata Internals), _or_ have this as subsection here (maybe the last one?), with a tittle like `Advanced: Reload Steps` -- as this can be useful to understand changes in performance, for instance, when that command is invoked?

Reply:
Ill make it a subsection.